src/pages/docs/wasi-http.mdx

```diff
@@ -0,0 +1,136 @@
+```
+
+Once the component is uploaded, we need to take note of the component id (fd13822e-b8fd-4c58-a87a-909feeea63ec) and version (0), we'll need them in the next step.
+
+## Defining the API
+
+The next step is to define the api. This is done by writing an [api definition](/docs/invoke/making-custom-apis) for our api.
+For our example this would look like this:
+
+```yaml
+id: example-api
+version: "0.1.0"
+draft: true
+routes:
+  - method: Get
+    path: "/api"
+    binding:
+      bindingType: http-handler
+      componentId:
+        componentId: "fd13822e-b8fd-4c58-a87a-909feeea63ec"
+        version: 0
+```
+
+Save the above to a file 'api.yaml'.
+
+## Uploading and deploying the API
+
+Next the api needs to be uploaded and deployed to Golem. This can be done using the CLI with the following two commands:
+
+1. This registers the api definition with Golem and makes it available for other commands:
+
+   ```bash
+   $ golem-cli api-definition add api.yaml
+
+   {
+     "id": "example-api",
+     "version": "0.1.0",
+     "routes": [
+       {
+         "method": "Get",
+         "path": "/api",
+         "security": null,
+         "binding": {
+           "componentId": {
+             "componentId": "fd13822e-b8fd-4c58-a87a-909feeea63ec",
+             "version": 0
+           },
+           "workerName": null,
+           "idempotencyKey": null,
+           "response": null,
+           "bindingType": "http-handler",
+           "responseMappingInput": null,
+           "workerNameInput": {
+             "types": {}
+           },
+           "idempotencyKeyInput": null,
+           "corsPreflight": null,
+           "responseMappingOutput": null
+         }
+       }
+     ],
+     "draft": true,
+     "createdAt": "2025-01-29T19:31:54.328610228Z"
+   }
+   ```
+
+2. Now we can deploy the api to our host (localhost in this example):
+
+   ```bash
+   $ golem-cli api-deployment deploy --definition example-api/0.1.0 --host localhost:9093
+
+   {
+     "apiDefinitions": [
+       {
+         "id": "example-api",
+         "version": "0.1.0"
+       }
+     ],
+     "site": {
+       "host": "localhost:9093",
+       "subdomain": null
+     },
+     "createdAt": "2025-01-29T19:31:54.369756Z"
+   }
+   ```
+
+## Invoking the API
+
+This concludes the process of uploading a component that implements the wasi:http/incoming-handler to Golem. The API is now being served
+and will respond to http requests on the configure route.
+
+```bash
+http localhost:9093/api
+
+"Hello, I'm a durable teapot."
+```
```

Comment:
Do we need to explain it with a better practical example ? That why do we need this done using API gateway?

Reply:
While most user uses the doc as a spec, if we are providing an example, it's good to be an example that shows the value-add of exposing http-handler binding type using a convincing example.

Reply:
:teapot:

Reply:
Let me do this as a followup, can't think of a good example right now